src/components/GenerateModal/UserCells.tsx

```diff
@@ -0,0 +1,91 @@
+import styled from 'styled-components';
+import {theme} from 'styles/Theme';
+import UserCell from './UserCell';
+import {User} from 'types/chatroom.types';
+import {useEffect, useState} from 'react';
+import LoadingCircle from 'components/LoadingCircle/LoadingCircle';
+
+interface UserCellsProps {
+  typed: string;
+  height: string;
+  $marginTop?: string;
+  $inputState?: string;
+  filteredUserData?: User[];
+  allocatedData: User[];
+  subData: User[];
+  onToggleUser: React.Dispatch<React.SetStateAction<[User[], User[]]>>;
+  searchUserData?: string;
+  isLoading?: boolean;
+}
+
+const UserCells = (props: UserCellsProps) => {
+  const [filteredUserData, setFilteredUserData] = useState<User[]>([]);
+
+  const {searchUserData, allocatedData, isLoading} = props;
+
+  useEffect(() => {
+    filterUsers();
+  }, [searchUserData, allocatedData]);
+
+  const filterUsers = () => {
+    if (!searchUserData) {
+      setFilteredUserData(allocatedData);
+    } else {
+      const filtered = allocatedData.filter(user => user.name.toLowerCase().includes(searchUserData.toLowerCase()));
+      setFilteredUserData(filtered);
+    }
+  };
```

Comment:
```suggestion
  const filterUsers = useCallback (() => {
    if (!searchUserData) {
      setFilteredUserData(allocatedData);
    } else {
      const filtered = allocatedData.filter(user => user.name.toLowerCase().includes(searchUserData.toLowerCase()));
      setFilteredUserData(filtered);
    }
  }, [searchUserData]);
```

이렇게 useCallback으로 함수 메모이제이션할 수 있어요